CDCgov/prime-reportstream
# Experience-7773: Add schema dropdown on validator page

This PR ...

This changeset adds a dropdown to the validator page to allow the user to select between schema options to validate against.  To accommodate this, I've added a `useSenderSchemaOptions` hook that always returns the three "master" schemas" in addition to the designated Sender schema if the current user is logged is as a Sender.

This also includes a decent amount of cleanup/rewrites because the logic in useFileHandler, FileHandler, and FileHandlerForm is oddly coupled together.  Will comment below!

To generate test files to validate against different schemas, you can use the CLI tool:
```
./prime data --input-fake <item_count> --input-schema <schema> --output <output_destination>
```

like:

```
./prime data --input-fake 10 --input-schema hl7/hcintegrations-covid-19 --output ~/Desktop/whatever.hl7
```

This seems to generate lots of duplicate columns so those have to be removed or validation will complain about those, but this is a starting point for test data!

Test Steps:
As an admin:
1. Log in
2. Go to http://localhost:3000/file-handler/validate
3. Verify that the three master schemas are available options
4. Upload a file to validate against the selected schemas --> ensure you get the right success/warning/error messages

As a Sender:
1. Log in
2. Go to http://localhost:3000/file-handler/validate
3. Verify that the three master schemas AND the Sender schema are available options
4. Upload a file to validate against the selected schemas --> ensure you get the right success/warning/error messages

This can also be tested against a logged-out user by changing [this line](https://github.com/CDCgov/prime-reportstream/blob/master/frontend-react/src/pages/Validate.tsx#L23) to be `export const ValidateWithAuth = Validate;` just to get around the authentication stuff.  The validate call will still work.

## Changes
As an admin or unauthenticated user:
<img width="1202" alt="Screenshot 2023-02-06 at 11 18 46" src="https://user-images.githubusercontent.com/2421042/217026752-3a3ceac0-419c-4acd-8172-94b4102f2aaa.png">

As a Sender:
<img width="1204" alt="Screenshot 2023-02-06 at 11 19 10" src="https://user-images.githubusercontent.com/2421042/217026753-b28c337d-7cf1-49c2-8457-41fa1617a170.png">

## Checklist

### Testing
<strike>- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?</strike>

## Linked Issues
- Fixes #7773

## To Be Done
- #8200
